### PR TITLE
Report Lookup plugin unit test coverage to Jenkins

### PR DESCRIPTION
**Pull request by Pooja-Gangwar**
### Desired Outcome

Generate test coverage reporting in CI, to ensure that the repo is sufficiently unit tested.

### Implemented Changes

Jenkins file and test_unit script file

### Connected Issue/Story

CyberArk internal issue link: [ONYX-15264](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-15264)

### Definition of Done
Implement code coverage reporting in CI, either with ccCoverage or ansible-test
Investigate whether the ccCoverage Jenkins plugin can be used for the Collection repo

#### Changelog

- [ ] The CHANGELOG has been updated, or
- [x] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [x] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -29,12 +29,26 @@ pipeline {
...
+            sh 'chmod +x ./tests/ansibletest.sh'
           }
         }
       }
     }
+    stage('Report Test Code Coverage'){
+          steps {
+            // dir('src/main/java'){
+            // ccCoverage('jacoco')
+            publishHTML (target : [allowMissing: false,
```
Comment (john-odonnell) [inline Jenkinsfile:43]: This coverage report looks great. After looking at the coverage report from this branch, we should try to eliminate reporting for test files & empty files (like `test_conjur_variable.py` and `__init__.py`), and try to report coverage on only our source files.

File: tests/output/coverage/units=controller=docker-default=python-3.8=coverage.67e1f73a4ab9.100.066554 [hunk 2]
```diff
@@ -0,0 +1 @@
+!coverage.py: This is a private format, don't read it directly!{"arcs":{}}
```
Comment (john-odonnell) [inline tests/output/coverage/units=controller=docker-default=python-3.8=coverage.67e1f73a4ab9.100.066554:1]: Files in the `tests/output/` directory shouldn't be checked into git - these coverage files should be created in Jenkins.

File: Jenkinsfile [hunk 3]
```diff
@@ -29,12 +29,26 @@ pipeline {
 
         stage("Test conjur_host_identity role") {
           steps {
-            sh './ci/test.sh -d conjur_host_identity'
-            junit 'roles/conjur_host_identity/tests/junit/*'
+            // sh './ci/test.sh -d conjur_host_identity'
+            // junit 'roles/conjur_host_identity/tests/junit/*'
+            sh 'chmod +x ./tests/ansibletest.sh'
```
Comment (john-odonnell) [inline Jenkinsfile:34]: It looks like you've disabled the `conjur_host_identity` role's tests here. Running the `tests/ansibletest.sh` script should happen elsewhere in the Jenkins pipeline.

File: .github/workflows/ansible-test.yml [hunk 4]
```diff
@@ -63,3 +63,54 @@ jobs:
...
+        run: pip install https://github.com/ansible/ansible/archive/${{ matrix.ansible }}.tar.gz --disable-pip-version-check
+
+      # Run the unit tests
+      - name: Run unit test
+        run: ansible-test units --docker default -v --python ${{ matrix.python }}
+        working-directory: ./ansible_collections/cyberark/conjur
+
+      # ansible-test support producing code coverage date
+      - name: Generate coverage report
+        run: ansible-test coverage html -v --requirements --group-by command --group-by version
```
Comment (john-odonnell) [inline .github/workflows/ansible-test.yml:110]: We shouldn't need to report coverage when the unit tests are run in GitHub Actions. If we report coverage to Jenkins, that should be enough.

File: tests/ansibletest.sh [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -eu
+
+mkdir -p ansible_collections/cyberark/
+cd ansible_collections/cyberark/
+git clone --single-branch --branch deleteit_1 https://github.com/cyberark/ansible-conjur-collection.git
```
Comment (john-odonnell) [inline tests/ansibletest.sh:5]: Why do we need to clone the repo here? If this script is run from Jenkins, the repo should be available to the script already.

**Review by john-odonnell — CHANGES_REQUESTED** [review]
Left a review below - this is a great start! We'll probably have to update this with updates to #108.

File: tests/ansibletest.sh (hunk 5 above)
Comment (Pooja-Gangwar) [reply tests/ansibletest.sh:5]: As per my understanding , the Ansible-test working directory should be like "/ansible_collections/{namespace}/{collection}/" but we dont have it . That's why I put our repo in this directory structure .

File: .github/workflows/ansible-test.yml (hunk 4 above)
Comment (Pooja-Gangwar) [reply .github/workflows/ansible-test.yml:110]: Sure John , I have removed report coverage from GitHub Actions .

File: Jenkinsfile (hunk 3 above)
Comment (Pooja-Gangwar) [reply Jenkinsfile:34]: Sure John,

File: tests/output/coverage/units=controller=docker-default=python-3.8=coverage.67e1f73a4ab9.100.066554 (hunk 2 above)
Comment (Pooja-Gangwar) [reply tests/output/coverage/units=controller=docker-default=python-3.8=coverage.67e1f73a4ab9.100.066554:1]: Done

File: Jenkinsfile (hunk 1 above)
Comment (Pooja-Gangwar) [reply Jenkinsfile:41]: Done

File: .github/workflows/ansible-test.yml (hunk 4 above)
Comment (Pooja-Gangwar) [reply .github/workflows/ansible-test.yml:110]: Done

File: Jenkinsfile (hunk 3 above)
Comment (Pooja-Gangwar) [reply Jenkinsfile]: Done

**Review by john-odonnell — APPROVED** [review]
LGTM 👍

**Pull request merged**